### PR TITLE
SITL parameters documentation

### LINKED ISSUE #22903: SITL parameters need documenting
## Feature request

**Is your feature request related to a problem? Please describe.**

We added support for documenting our SITL parameters in our Wiki and GCS-consumed XML.

However, the parameters are not yet documented.  A whitelist was created for the existing SITL parameters - so new parameters need to be documented, older parameters get a free-pass.  This is the whitelist: https://github.com/ardupilot/ardupilot/blob/master/Tools/autotest/common.py#L2129

We should remove entries from that whitelist so we get parameter docs in places like MissionPlanner's parameter screen while running simulations.

Parameters are documented in https://github.com/ardupilot/ardupilot/blob/master/libraries/SITL/SITL.cpp#L551

Note that this can be a gradual process - even a handful every now and then will whittle away at the list.

**Describe the solution you'd like**

Parameters documented and removed from the whitelist.

**Describe alternatives you've considered**

Leaving whitelist in place and forgetting about the problem.

**Platform**
[ X ] All
[  ] AntennaTracker
[  ] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

### PR DESCRIPTION
**Pull request by gmazzucchi**
Addresses part of #22903

### CODE REVIEW

**Comment by magicrub** [comment]
Nice work!

**Comment by tzarjakob** [comment]
when it's ready to merge, I'll squash all the commits into one according to the contribution style

**Comment by tzarjakob** [comment]
After your final check, this is ready for merge.

File: libraries/SITL/SITL.cpp [hunk 1]
```diff
@@ -563,37 +610,97 @@ const AP_Param::GroupInfo SIM::var_sfml_joystick[] = {
...
+    // @Param: ACC2_RND
+    // @DisplayName: ACC2 Random
+    // @Description: Simulated Noise of ACCEL2
+    // @Units: m/s/s
     AP_GROUPINFO("ACC3_RND",     13, SIM, accel_noise[2], 0),
 #endif
+    // @Param: GYR1_SCALE
+    // @DisplayName: GYR1 Scale
+    // @Description: GYR1 scale factor
+    // @Units: %
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:690]: Is this *really* a percent?

**Review by peterbarker — CHANGES_REQUESTED** [review]
Pretty sure that's not a percentage.  Prove me wrong :-)

There's some syntax error in your changes - the Python routines used to extract this parameter metadata (so that it can be used in the Wiki and in the ground control stations) is unhappy.  Check the CI results and if you can't work out what's going wrong ping me on Discord.

File: libraries/SITL/SITL.cpp (hunk 1 above)
Comment (tzarjakob) [reply libraries/SITL/SITL.cpp:690]: You are right. My mistake, the gyroscope scale is measured in rad/sec. I'll also change the misleading comment on https://github.com/ArduPilot/ardupilot/blob/b638a3576b1a1d397933891c6512c4a11e990f95/libraries/SITL/SITL.h#L496

**Comment by tzarjakob** [comment]
Thank you for the feedback. I checked the logs and some colons were actually missing.

File: libraries/SITL/SITL.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/SITL/SITL.cpp]: Mmmmm... I think you will find these need to be unit-less for it to be correct units-wise.

If the scale was radians then you'd have radians-per-second/radians-per-second which would mean the result would be unit-less - but we know that the gyro data has to be in radians-per-second....

Scales are usually unit-less.

I've just had a look - turns out I'm wrong, these *are* percentages!

https://github.com/ardupilot/ardupilot/blob//libraries/AP_InertialSensor/AP_InertialSensor_SITL.cpp#L283

Apologies for the run-around.  If you could add a comment on that line 496 saying, "look in AP_InertialSensor_SITL for details" that would be good.

My bad.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Tests need to pass

**Comment by peterbarker** [comment]
... don't scope-creep yourself!  Plenty of other people will do that for you :-)

We're not running out of PR numbers, so let's get this one in and then do more if you'd like...

**Comment by peterbarker** [comment]
Argh!  Really sorry, this one managed to slip through the cracks.  I've improved my procedures a little to stop that happening.

As luck would have it, these ones are now documented in master.

Many thanks for making the effort here - this should have been merged :-/

**Pull request closed without merging**